### PR TITLE
Majsylw clean

### PR DESCRIPTION
**Pull request by majsylw**
simplify symmetry operator, switch to one for loop

### CODE REVIEW

File: parse-hamnosys.py [hunk 1]
```diff
@@ -157,56 +157,33 @@ def main(args):
...
-            continue
-        elif char == SymmetryOperatorsDict["8"]:
-            data.at[index, "Symmetry operator"] = 8
-            Hamnosys_copy[index] = Hamnosys_copy[index][1:]
-            continue
-        # Check if non dominan hand is analyzed as only and set flag
-        elif char == SymmetryOperatorsDict["9"]:
-            data.at[index, "Symmetry operator"] = 0
-            Hamnosys_copy[index] = Hamnosys_copy[index][1:]
-            data.at[index, "NonDom first"] = 1
```
Comment (mplantykow) [inline parse-hamnosys.py:206]: Do we really want to get rid of this column?

File: parse-hamnosys.py [hunk 2]
```diff
@@ -181,17 +173,15 @@ def main(args):
...
+        for i in range(len(UnknownSymbols1Dict)):
             char = Hamnosys_copy[index][0]
             for key, value in UnknownSymbols1Dict.items():
                 if char == value:
                     Hamnosys_copy[index] = Hamnosys_copy[index][1:]
 
-    # Dominan hand base for is expected here,
-    # so we iter trough all possible options
-    for index, row in data.iterrows():
+        # Dominan hand base for is expected here,
```
Comment (mplantykow) [inline parse-hamnosys.py]: btw we got a typo in here (Dominan**t**)

File: parse-hamnosys.py [hunk 3]
```diff
@@ -154,23 +154,14 @@ def main(args):
     data["Dominant - Handshape - Bending"] = 0
 
     for index, _ in data.iterrows():
-        # Search for symmetry operators that consists of 3 or 2 symbols,
+        # Search for symmetry operators that consists of from 3 to 1 symbol,
         # remove if found
-        for i in range(1, 7):
-            j = 3 if i < 3 else 2
+        for i in range(1, 10):
+            j = 3 if i < 3 else (2 if i < 7 else 1)
```
Comment (mplantykow) [inline parse-hamnosys.py:144]: Nice

File: parse-hamnosys.py [hunk 4]
```diff
@@ -153,122 +153,75 @@ def main(args):
...
-            Hamnosys_copy[index] = Hamnosys_copy[index][1:]
-            data.at[index, "NonDom first"] = 1
-            continue
+        for i in range(1, 10):
+            j = 3 if i < 3 else (2 if i < 7 else 1)
+            char = Hamnosys_copy[index][0:j]
+            if set(char) == set(SymmetryOperatorsDict[str(i)]):
+                data.at[index, "Symmetry operator"] = i % 9
+                Hamnosys_copy[index] = Hamnosys_copy[index][j:]
+                continue
```
Comment (majsylw) [inline parse-hamnosys.py:152]: @mplantykow look here - I've just shortened all logic for Symmetry in one loop - range 1-9 so for all 9 HamNoSys signs

**Review by mplantykow — APPROVED** [review]
We should consider having non domiant only as a sparate class like in the past.
Edit: flake8 returns many syle related errors, consider fixing those.

File: parse-hamnosys.py (hunk 4 above)
Comment (mplantykow) [reply parse-hamnosys.py:152]: 9 is not a symmetry operator. Before your change was introduced 9 was analyzed in a separate if and if found 1 was assigned to column named "NonDom first", not to symmetry operators.

**Review by MilenaOlech — APPROVED** [review]

**Pull request merged**